### PR TITLE
[core] Move `rcore.h` content to inside `rcore.c`

### PR DESCRIPTION
**Pull request by ubkp**
### Changes
1. Moves `rcore.h` content to inside `rcore.c`:

| Content | From `rcore.h` | To `rcore.c` |
| :--- | :--- | :--- |
| Includes | [L40-L51](https://github.com/raysan5/raylib/pull/3479/files#diff-7341b65fe444443a67b5c67d194f97dd0c919b63c579f15ce018de206cf84864L40-L51) | [R88-R94](https://github.com/raysan5/raylib/pull/3479/files#diff-39fea0e3daa23741e81daf967717f35c7842426aab26a0bc3f525d6a2f03e339R88-R94), [R99-R100](https://github.com/raysan5/raylib/pull/3479/files#diff-39fea0e3daa23741e81daf967717f35c7842426aab26a0bc3f525d6a2f03e339R99-R100) |
| Defines and Macros | [L53-L105](https://github.com/raysan5/raylib/pull/3479/files#diff-7341b65fe444443a67b5c67d194f97dd0c919b63c579f15ce018de206cf84864L53-L105) | [R176-R228](https://github.com/raysan5/raylib/pull/3479/files#diff-39fea0e3daa23741e81daf967717f35c7842426aab26a0bc3f525d6a2f03e339R176-R228) |
| Types and Structures | [L107-L206](https://github.com/raysan5/raylib/pull/3479/files#diff-7341b65fe444443a67b5c67d194f97dd0c919b63c579f15ce018de206cf84864L107-L206) | [R230-R329](https://github.com/raysan5/raylib/pull/3479/files#diff-39fea0e3daa23741e81daf967717f35c7842426aab26a0bc3f525d6a2f03e339R230-R329) |
| Global Variable | [L211](https://github.com/raysan5/raylib/pull/3479/files#diff-7341b65fe444443a67b5c67d194f97dd0c919b63c579f15ce018de206cf84864L211) | [Removed](https://github.com/raysan5/raylib/pull/3479#discussion_r1375491939) |

### Reference

https://github.com/raysan5/raylib/pull/3475#issuecomment-1784143084

### Environment
Tested on:
- `PLATFORM_DESKTOP` on Linux (Ubuntu 22.04 64-bit).
- `PLATFORM_DESKTOP_SDL` on Linux (Ubuntu 22.04 64-bit).
- `PLATFORM_WEB` on Firefox (115.3.1esr 64-bit) and Chromium (117.0.5938.149 64-bit) both on Linux (Mint 21.1 64-bit).
- `PLATFORM_DRM` on Raspberry Pi 3 Model B 1.2 with Linux (Raspberry Pi OS 11 Bullseye 32-bit armhf).

### Changes
**1:** added line marks; added `SDL` environment.
**2:** formatting.
**3:** formatting; added `DRM` environment.
**4:** updating.

### CODE REVIEW

File: src/rcore.c [hunk 1]
```diff
@@ -166,11 +173,167 @@ __declspec(dllimport) int __stdcall WideCharToMultiByte(unsigned int cp, unsigne
...
+
+    } Time;
+} CoreData;
+
 //----------------------------------------------------------------------------------
 // Global Variables Definition
 //----------------------------------------------------------------------------------
 RLAPI const char *raylib_version = RAYLIB_VERSION;  // raylib version exported symbol, required for some bindings
 
+extern CoreData CORE;
```
Comment (raysan5) [inline src/rcore.c:336]: I think this line shouldn't be required

**Comment by raysan5** [comment]
@ubkp thank you very much for the improvement! I just added a small review...

File: src/rcore.c (hunk 1 above)
Comment (ubkp) [reply src/rcore.c]: Removed :+1:

**Comment by raysan5** [comment]
@ubkp thank you very much again for the review!

**Pull request merged**